gateway/gateway.go

```diff
@@ -59,12 +63,41 @@ func New(rootLogger *logrus.Logger, cfg *config.Gateway, r *Router) *Gateway {
 	if err := gw.AddConfig(cfg); err != nil {
 		logger.Errorf("Failed to add configuration to gateway: %#v", err)
 	}
+
+	persistentMessageStorePath, usePersistent := gw.Config.GetString("PersistentMessageStorePath")
```

Comment:
Am I getting the config correctly here?
Unless I set it at the very top outside of any nesting it doesn't seem to work.

Reply:
> @yousefmansy1 I'm merging them, thanks for your work on those PRs and sorry for the delays.

Thanks @42wim!
No worries at all, I'm sure you had a busy schedule.

As for how we're using getting config values, I have to put this value at a top level or else value comes back as null from when I call `gw.Config.GetString("PersistentMessageStorePath")`
eg: 
```
PersistentMessageStorePath="/etc/matterbridge/badger"

[discord]
	[discord.bot]
		...

[telegram]
	[telegram.bot]
		...

[whatsapp]
	[whatsapp.googlevoice1]
		...

[general]
	...
```

Reply:
Resolved in 60219a39d1624df64de9421177b326bde269319f

Let me know if you hate the formatting applied in config, and I can undo.